lib/s3routes/routes/routeWebsite.ts

```diff
@@ -30,7 +30,7 @@ export default function routerWebsite(
                 // request being redirected
                 if (redirectInfo) {
                     if (err && redirectInfo.withError) {
-                        return routesUtils.redirectRequestOnError(err,
+                        return routesUtils.redirectRequestOnError(err as ArsenalError,
```

Comment:
do you know why we need type coercion here?
`callApiMethod` already defines that the callback receives an `ArsenalError` error...

Reply:
It's because the error we can currently get from APIs is defined in

```ts

export type CallApiMethod = (
    methodName: string,
    request: http.IncomingMessage,
    response: http.ServerResponse,
    log: RequestLogger,
    callback: (err: ArsenalError | Error | null, ...data: any[]) => void,
) => void;
```

So it can be a non-arsenal error, this change was made on purpose so we benefit from ts checks everywhere and cover all paths that could lead to errors. The case of website is special, we already know it can only be arsenal errors. All of this will be removed once we have the "long term" solution in place

Reply:
indeed, I missed the alternate types 🤦‍♂️

but it seems a bit dangerous to just "expect" these will be arsenal errors and blindly trust the cast : if we want to rely on typescript to detect errors, best do it all the way and add an early check that it is indeed an ArsenalError (or just don't cast, and allow `redirectRequestOnError` to handle any `Error`)

Reply:
I updated the function so that it accepts non-arsenal errors - i kept the logic to dynamically recreate an arsenal error in this case, so we minimize the changes (we still need to return an Arsenal error today to the client).

Reply:
👍, but then we should remove the type-coercion altogether

Reply:
yes... thanks